adcc/ElectronicTransition.py

```diff
@@ -0,0 +1,269 @@
+            absorption = self.cross_section
+            ylabel = "Cross section (au)"
+        elif yaxis in ["rot", "rotational_strength", "rotatory_strength"]:
+            absorption = self.rotatory_strength
+            ylabel = "Rotatory strength (au)"
+        else:
+            raise ValueError("Unknown yaxis specifier: {}".format(yaxis))
+
+        sp = ExcitationSpectrum(energies, absorption)
+        sp.xlabel = xlabel
+        sp.ylabel = ylabel
+        if not broadening:
+            plots = sp.plot(style="discrete", **kwargs)
+        else:
+            kwdisc = kwargs.copy()
+            kwdisc.pop("label", "")
+            plots = sp.plot(style="discrete", **kwdisc)
+
+            kwargs.pop("color", "")
+            sp_broad = sp.broaden_lines(width, shape=broadening)
+            plots.extend(sp_broad.plot(color=plots[0].get_color(),
+                                       style="continuous", **kwargs))
+
+        if xaxis in ["nm"]:
+            # Invert x axis
+            plt.xlim(plt.xlim()[::-1])
+        return plots
+
+
+class State2StateTransition(ElectronicTransition):
+    """
+    Documentation
+    """
+    def __init__(self, parent_state, initial=0, final=None):
+        self.parent_state = parent_state
+        self.reference_state = self.parent_state.reference_state
+        self.ground_state = self.parent_state.ground_state
+        self.matrix = self.parent_state.matrix
+        self.property_method = self.parent_state.property_method
+        self.operators = self.parent_state.operators
+
+        self.initial = initial
+        self.final = final
+
+        self.initial_excitation_vector = \
+            self.parent_state.excitation_vector[self.initial]
+        if self.final is None:
+            other_excitation_energy = np.delete(
+                self.parent_state.excitation_energy.copy(), self.initial
+            )
+        else:
+            other_excitation_energy = \
+                np.array([self.parent_state.excitation_energy[self.final]])
+        self.excitation_energy = other_excitation_energy -\
+            self.parent_state.excitation_energy[self.initial]
+    # TODO: describe?!
+
+    @cached_property
+    @mark_excitation_property(transform_to_ao=True)
+    @timed_member_call(timer="_property_timer")
+    def transition_dm(self):
+        """
+        List of transition density matrices from
+        initial state to final state/s
+        """
+        # TODO: only states above self.initial
+        if self.final is None:
+            return [
+                adc_pp.state2state_transition_dm(self.property_method,
+                                                 self.ground_state,
+                                                 self.initial_excitation_vector,
+                                                 evec,
+                                                 self.matrix.intermediates)
+                for i, evec in enumerate(self.parent_state.excitation_vector)
```

Comment:
That's exactly the part I hate about black. I think that is too many lines for such a call. Can you try to compress that?

Reply:
Done.